File: js/platform/src/modules/metering.ts
import { http } from "../core/http.js";
import { Endpoints } from "../core/endpoints.js";
import type { ApiResult } from "../core/result.js";

export const metering = {
  getBudget(): Promise<ApiResult<Record<string, unknown>>> {
    return http.getMap(Endpoints.meteringBudget);
  },

  getUsage(query?: { period?: string }): Promise<ApiResult<Record<string, unknown>>> {
    return http.getMap(Endpoints.meteringUsage, query);
  },
};

File: flutter/platform/lib/src/modules/metering.dart
import '../core/http_client.dart';
import '../core/api_endpoints.dart';
import '../core/result.dart';

/// Metering module — budget status and usage.
class MeteringModule {
  const MeteringModule();

  Future<ApiResult<Map<String, dynamic>>> getBudget() =>
      HttpClient.instance.getMap(ApiEndpoints.meteringBudget);

  Future<ApiResult<Map<String, dynamic>>> getUsage({String? period}) =>
      HttpClient.instance.getMap(
        ApiEndpoints.meteringUsage,
        queryParams: {if (period != null) 'period': period},
      );
}
